### PR TITLE
CCIP-4359 fee quoter supports interface for keystone

### PR DESCRIPTION
**Pull request by 0xsuryansh**
Keystone Forwarder checks for the receiver to support IReceiver interface using ERC165 supportsInterface

This PR :
- Enforced IReceiver implementations to implement supportsInterface

### CODE REVIEW

**Review by DeividasK — DISMISSED** [review]

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 1]
```diff
@@ -3,6 +3,8 @@ pragma solidity 0.8.24;
 
 import {IReceiver} from "../keystone/interfaces/IReceiver.sol";
 import {ITypeAndVersion} from "../shared/interfaces/ITypeAndVersion.sol";
+
+import {IERC165} from "../vendor/openzeppelin-solidity/v5.0.2/contracts/interfaces/IERC165.sol";
```
Comment (RensR) [inline contracts/src/v0.8/ccip/FeeQuoter.sol]: import is third party, should go at the bottom

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 2]
```diff
@@ -493,6 +495,14 @@ contract FeeQuoter is AuthorizedCallers, IFeeQuoter, ITypeAndVersion, IReceiver,
     }
   }
 
+  /// @notice Signals which version of the pool interface is supported
+  function supportsInterface(
```
Comment (RensR) [inline contracts/src/v0.8/ccip/FeeQuoter.sol:498]: should we not use 

`interfaceId == type(IFeeQuoter).interfaceId || super.supportsInterface` or something similar?

File: contracts/src/v0.8/keystone/interfaces/IReceiver.sol [hunk 3]
```diff
@@ -1,8 +1,10 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
+import {IERC165} from "../../vendor/openzeppelin-solidity/v5.0.2/contracts/utils/introspection/IERC165.sol";
+
 /// @title IReceiver - receives keystone reports
-interface IReceiver {
+interface IReceiver is IERC165 {
```
Comment (RensR) [inline contracts/src/v0.8/keystone/interfaces/IReceiver.sol:8]: please add a comment that it's important to support IReceiver or no reports would be sent

File: contracts/src/v0.8/ccip/FeeQuoter.sol (hunk 2 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/FeeQuoter.sol:498]: FeeQuoter is IReceiver, there's no implementation to inherit from

**Review by DeividasK — DISMISSED** [review]

File: contracts/src/v0.8/ccip/FeeQuoter.sol [hunk 4]
```diff
@@ -3,6 +3,7 @@ pragma solidity 0.8.24;
 
 import {IReceiver} from "../keystone/interfaces/IReceiver.sol";
 import {ITypeAndVersion} from "../shared/interfaces/ITypeAndVersion.sol";
+
```
Comment (RensR) [inline contracts/src/v0.8/ccip/FeeQuoter.sol]: nit: remove newline

**Review by RensR — COMMENTED** [review]
can you add a test that actually uses the keystone contract to call into the feeQuoter

File: contracts/src/v0.8/ccip/FeeQuoter.sol (hunk 4 above)
Comment (0xsuryansh) [reply contracts/src/v0.8/ccip/FeeQuoter.sol]: https://github.com/smartcontractkit/chainlink/pull/15448/commits/e4da55e729f25dacc07c93c5bbb00b00676a3a53

**Review by RensR — APPROVED** [review]

**Review by kalverra — APPROVED** [review]

**Review by DeividasK — APPROVED** [review]

**Pull request merged**